public/src/overmind/actions.tsx

```diff
@@ -190,84 +215,86 @@ export const updateEnrollment = async ({ state, effects }: Context, { enrollment
         case Enrollment_UserStatus.TEACHER:
             confirmed = confirm(`Are you sure you want to promote ${enrollment.user.Name} to teacher status?`)
             break
+        case Enrollment_UserStatus.PENDING:
+            // Status pending should never be set by this function.
+            // If the intent is to accept a pending enrollment, status should be set to student.
+            return
+    }
+    if (!confirmed) {
+        return
     }
 
-    if (confirmed) {
-        // Lookup the enrollment
-        // The enrollment should be in state, if it is not, do nothing
-        const enrollments = state.courseEnrollments[state.activeCourse.toString()] ?? []
-        const found = enrollments.findIndex(e => e.ID === enrollment.ID)
-        if (found === -1) {
-            return
-        }
+    // Lookup the enrollment
+    // The enrollment should be in state, if it is not, do nothing
+    const enrollments = state.courseEnrollments[state.activeCourse.toString()] ?? []
+    const found = enrollments.findIndex(e => e.ID === enrollment.ID)
+    if (found === -1) {
+        return
+    }
 
-        // Clone enrollment object and change status
-        const temp = enrollment.clone()
-        temp.status = status
+    // Clone enrollment object and change status
+    const temp = enrollment.clone()
+    temp.status = status
 
-        // Send updated enrollment to server
-        const response = await effects.grpcMan.updateEnrollments([temp])
-        if (success(response)) {
-            // If successful, update enrollment in state with new status
-            if (status === Enrollment_UserStatus.NONE) {
-                // If the enrollment is rejected, remove it from state
-                enrollments.splice(found, 1)
-            } else {
-                // If the enrollment is accepted, update the enrollment in state
-                enrollments[found].status = status
-            }
+    // Send updated enrollment to server
+    const response = await effects.api.client.updateEnrollments({ enrollments: [temp] })
+    if (success(response)) {
```

Comment:
Seems to break the pattern (now adopted elsewhere) of checking for `if (response.error)` and returning early and keeping the "happy path" non-indented below the error-guard.